namespaces.go

```diff
@@ -30,7 +36,12 @@ type Namespace struct {
 // CreateNamespace creates a new Namespace. Returns error if could not be created
 // or another namespace already exists.
 func (h *Headscale) CreateNamespace(name string) (*Namespace, error) {
+	var err error
 	namespace := Namespace{}
+	name, err = NormalizeNamespaceName(name)
```

Comment:
I do not think we should normalize what users pass to functions, it should be catched and validated _before_ it reaches this function. This is can cause unexpected behaviour for the dev, and it should not need to be done before methods like delete.

We should validate input to the `headscale namespace create` command, not change it, but rather return an error.

And we should normalize it in the OIDC logic, where we _know_ its not valid.

Reply:
That's was my initial thought, I'll go back there !

Reply:
So, I didn't implemented in the client but server-side. Because if we use another mean to talk to the API, the check would not be implemented.